jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/jobs/coursepublish/CourseSitePublishJob.java

```diff
@@ -0,0 +1,229 @@
+ * limitations under the License.
+ */
+package org.sakaiproject.component.app.scheduler.jobs.coursepublish;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.StatefulJob;
+import org.sakaiproject.component.api.ServerConfigurationService;
+import org.sakaiproject.coursemanagement.api.CourseSitePublishService;
+import org.sakaiproject.tool.api.Session;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.user.api.User;
+import org.sakaiproject.user.api.UserDirectoryService;
+import org.sakaiproject.user.api.UserNotDefinedException;
+
+
+/**
+ * quartz job to remove course sites after a specified period of time.
+ */
+public class CourseSitePublishJob implements StatefulJob {
+
+   // logger
+   private final transient Log logger = LogFactory.getLog(getClass());
+
+   // sakai.properties
+   public final static String PROPERTY_COURSE_SITE_PUBLISH_USER                             = "course_site_publish_service.user";
+   public final static String PROPERTY_COURSE_SITE_PUBLISH_NUM_DAYS_BEFORE_TERM_STARTS      = "course_site_publish_service.num_days_before_term_starts";
+
+   // default values for the sakai.properties
+   public final static String DEFAULT_VALUE_COURSE_SITE_PUBLISH_USER                        = "admin";
+   public final static int DEFAULT_VALUE_COURSE_SITE_PUBLISH_NUM_DAYS_BEFORE_TERM_STARTS = 14;
+
+   // sakai services
+   private CourseSitePublishService   courseSitePublishService;
+   private ServerConfigurationService serverConfigurationService;
+   private SessionManager sessionManager;
+   private UserDirectoryService userDirectoryService;
+
+   // data members
+   private User user;                      // sakai user who will be used when running this job
+   private int numDaysBeforeTermStarts;    // number of days before a term starts when course sites will be published and made available to students enrolled in the course.
+
+
+
+   /**
+    * default constructor.
+    */
+   public CourseSitePublishJob() {
+      // no code necessary
+	}
+
+   /**
+    * called by the spring framework.
+    */
+   public void destroy() {
+      logger.info("destroy()");
+
+      // no code necessary
+   }
+
+   /**
+    * called by the spring framework after this class has been instantiated, this method reads in the following values from sakai.properties:
+    * <ol>
+    *    <li>course_site_publish_service.user                    </li>
+    *    <li>course_site_publish_service.num_days_before_term_starts</li>
+    * </ol>
+    */
+   public void init() {
+
+      logger.debug("init()");
+
+      // get the number of days after a term ends after which course sites that have expired will be removed
+      String numDaysBeforeTermStartsString = serverConfigurationService.getString(PROPERTY_COURSE_SITE_PUBLISH_NUM_DAYS_BEFORE_TERM_STARTS);
```

Comment:
The difference here is that getInt, first try to view if the property exists or not. If not, then try to return the default value, and then checks if the number is valid.  In that case, if the property exists but the number in the property is not valid, it launch an exception.
This way, first check that the value is valid and if not then use the default.